src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/MessageboxInstancesController.cs

```diff
@@ -172,20 +178,18 @@ public async Task<ActionResult> GetMessageBoxInstance(int instanceOwnerPartyId,
                 return BadRequest("Unable to perform query.");
             }
 
-            List<InstanceEvent> result = await _instanceEventRepository.ListInstanceEvents(instanceId, eventTypes, null, null);
-
-            // filtering out Create & delete data element event
-            result = result.Where(r => string.IsNullOrEmpty(r.DataId) || r.EventType.Equals(InstanceEventType.Saved)).ToList();
```

Comment:
I cant see any filtering be done in the repository class.. This means all events will be exposed in the activity log. With auto save enabled in app frontend, this means way more events than it is suitable to show to the end user.

Reply:
Added some filtering, but it's very basic right now.